test/e2e/scrapeconfig_test.go

```diff
@@ -226,3 +238,88 @@ func testPromOperatorStartsWithoutScrapeConfigCRD(t *testing.T) {
 	_, err = framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, []string{ns}, nil, []string{ns}, nil, false, true, true)
 	require.NoError(t, err)
 }
+
+// testScrapeConfigKubernetesNodeRole tests whether Kubernetes node monitoring works as expected
+func testScrapeConfigKubernetesNodeRole(t *testing.T) {
+	skipPrometheusTests(t)
+
+	testCtx := framework.NewTestCtx(t)
+	defer testCtx.Cleanup(t)
+	ns := framework.CreateNamespace(context.Background(), t, testCtx)
+	framework.SetupPrometheusRBAC(context.Background(), t, testCtx, ns)
+
+	_, err := framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, []string{ns}, nil, []string{ns}, nil, false, true, true)
+	require.NoError(t, err)
+
+	p := framework.MakeBasicPrometheus(ns, "prom", "group", 1)
+	p.Spec.ScrapeConfigSelector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"role": "scrapeconfig",
+		},
+	}
+	_, err = framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, p)
+	require.NoError(t, err)
+
+	// For prometheus to be able to scrape nodes it needs to able to authenticate
+	// using mTLS certificates issued for the ServiceAccount "prometheus"
+	secretName := "scraping-tls"
+	createServiceAccountSecret(t, "prometheus", ns)
+	createmTLSSecret(t, secretName, ns)
+
+	sc := framework.MakeBasicScrapeConfig(ns, "scrape-config")
+	sc.Spec.Scheme = pointer.String("HTTPS")
+	sc.Spec.Authorization = &monitoringv1.SafeAuthorization{
+		Credentials: &v1.SecretKeySelector{
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: "prometheus-sa-secret",
+			},
+			Key: "token",
+		},
+	}
+	sc.Spec.TLSConfig = &monitoringv1.SafeTLSConfig{
+		InsecureSkipVerify: true,
+		CA: monitoringv1.SecretOrConfigMap{
+			Secret: &v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: "ca.crt",
+			},
+		},
+		Cert: monitoringv1.SecretOrConfigMap{
+			Secret: &v1.SecretKeySelector{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: secretName,
+				},
+				Key: "cert.pem",
+			},
+		},
+		KeySecret: &v1.SecretKeySelector{
+			LocalObjectReference: v1.LocalObjectReference{
+				Name: secretName,
+			},
+			Key: "key.pem",
+		},
+	}
+
+	sc.Spec.KubernetesSDConfigs = []monitoringv1alpha1.KubernetesSDConfig{
+		{
+			Role: "Node",
+		},
+	}
+	_, err = framework.CreateScrapeConfig(context.Background(), ns, sc)
+	require.NoError(t, err)
+
+	// Check that the targets appear in Prometheus and does proper scrapping
+	if err := framework.WaitForHealthyTargets(context.Background(), ns, "prometheus-operated", 1); err != nil {
+		t.Fatal(err)
+	}
+
+	// Remove the ScrapeConfig and check that the targets disappear in Prometheus
```

Comment:
```suggestion
	// Remove the ScrapeConfig
```

You're checking that the targets disappeared in the next step, right? Let's not duplicate the comments :P

Reply:
aah good catch :D